plotly/tools.py

```diff
@@ -1296,3 +1304,686 @@ def return_figure_from_figure_or_data(figure_or_data, validate_figure):
             )
 
     return figure
+
+
+class TraceFactory(dict):
```

Comment:
I think the inheritance pattern between TraceFactory and _Quiver is a little strange.

rather?

```
class TraceFactory(object):
    # stuff
```

and

```
# in _Quiver
TraceFactory.validate(*stuff, *stuffs)
```

Since it's a static method, you can just call validate from `TraceFactory`.